src/NHibernate/Criterion/ProjectionsExtensions.cs

```diff
@@ -21,6 +21,17 @@ public static class ProjectionsExtensions
 			return Projections.Alias(projection, aliasContainer);
 		}
 
+		/// <summary>
+		/// Create an alias for a projection
+		/// </summary>
+		/// <param name="projection">the projection instance</param>
+		/// <param name="alias">LambdaExpression returning an alias</param>
```

Comment:
@fredericDelaporte Yep. It's accidentally leaked here. But can I keep it? It's convenient extension for using  with `nameof` (which I think should be preferred over lambda expression):
`Projections.RootEntity().WithAlias(nameof(r.Root))`
vs
`Projections.RootEntity().WithAlias(() => r.Root)`

Though it's easily can be placed directly in user code - I'm not going to push it :) .

Reply:
I think it can stay.

About lambda vs `nameof`, I agree. Almost all QueryOver API is somewhat obsoleted by `nameof`, since it allows using the Criteria API without magic strings.